cmd/tt/bucket.go

```diff
@@ -375,67 +446,87 @@ These 'push' commands result in the following bucket structures.
 			cmd.Fatal(fmt.Errorf("thread is not defined"))
 		}
 
-		var names []string
-		var paths []string
-		bucketPath, args := args[len(args)-1], args[:len(args)-1]
-		for _, a := range args {
-			abs, err := filepath.Abs(a)
-			if err != nil {
+		buck, err := local.NewBucket(root, options.BalancedLayout)
+		if err != nil {
+			cmd.Fatal(err)
+		}
+		diff := getDiff(buck, root)
+		force, err := c.Flags().GetBool("force")
+		if err != nil {
+			cmd.Fatal(err)
+		}
+		if force {
+			// Reset the archive to just the seed file
+			seed := filepath.Join(root, bucks.SeedName)
+			ctx, acancel := context.WithTimeout(context.Background(), cmdTimeout)
+			defer acancel()
+			if err = buck.ArchiveFile(ctx, seed, bucks.SeedName); err != nil {
 				cmd.Fatal(err)
 			}
-			if !strings.HasPrefix(abs, root) {
-				cmd.Fatal(fmt.Errorf("the path %s is not under the current bucket root", abs))
-			}
-
-			dir := filepath.Dir(a)
-			if err := filepath.Walk(a, func(n string, info os.FileInfo, err error) error {
-				if err != nil {
-					cmd.Fatal(err)
-				}
-				if !info.IsDir() {
-					if strings.HasSuffix(n, ".DS_Store") {
-						return nil
+			// Add unique additions
+		loop:
+			for _, c := range getDiff(buck, root) {
+				for _, x := range diff {
+					if c.Path == x.Path {
+						continue loop
 					}
-					names = append(names, n)
-					var p string
-					if n == a { // This is a file given as an arg
-						// In this case, the bucket path should not include the directory
-						p = filepath.Join(bucketPath, info.Name())
-					} else { // This is a directory given as an arg, or one of its sub directories
-						// The bucket path should maintain directory structure
-						if dir != "." {
-							n = strings.TrimPrefix(n, dir)
-						}
-						p = filepath.Join(bucketPath, n)
-					}
-					paths = append(paths, p)
 				}
-				return nil
-			}); err != nil {
-				cmd.Fatal(err)
+				diff = append(diff, c)
```

Comment:
Took me a minute to remember the logic here. If we just use the diff from the reset archive, it will report all change paths as _additions_, even if the original diff (our staged changes) included modifications or deletions for those paths. So, we just walk the reset diff, only adding the change if there isn't already one present for the given path.